### ISSUE TITLE
md-snackbar (toast)

### ISSUE THREAD

**Issue by jelbourn** [angular/material2#115]
https://www.google.com/design/spec/components/snackbars-toasts.html

Depends on #113

**Comment by jelbourn** [comment]
Status update:
[Design doc](https://docs.google.com/document/d/1XuD-sfGjZRB-TVFMULmD_FOSsOVJACVRy3XRJsQyYeQ/preview)
[Proof of concept](https://jperrott-material2.firebaseapp.com/snack-bar)

**Comment by jelbourn** [comment]
Closing this since we have an initial version

**Comment by AlvaroOrduna** [comment]
As Material Design specification says, a SnackBar may or may not have an action. There any way to add a SnackBar without any action for now?

@angular/material 2.0.0.alpha.9-3

**Comment by jammerware** [comment]
@AlvaroOrduna I was able to add an actionless SnackBar by passing `null` as the `actionLabel` argument to `snackBar.open`.

**Comment by mhmo91** [comment]
Duration doesn't work with me, it doesn't even exist in the config class
i'm using version    "@angular/material": "^2.0.0-alpha.10",
Also, how about the color of the toaster? like danger, warning, success... thankss

**Comment by jammerware** [comment]
@mhmo91 I don't think build with the duration property has been pushed to the npm feed yet. [The commit](https://github.com/angular/material2/commit/6ce63962ec4d9dc5ac8ebdb0ec83f17335b54334) in question was only made just under a couple of weeks ago.

**Comment by cheresier** [comment]
I have a suspicion that the viewContainerRef setting of MdSnackBarConfig is working correctly. I have a component (let's call it CompanyComponent) that consists of one MdCard with a button inside of it.  Pressing the button opens a MdSnackBar. Since each card has a button that opens a snack bar I want the snack bar to be anchored to the bottom of each card. Multiple instances of the CompanyComponent appear in the parent, AppComponent. I open the snack bar using the following code:

```
    @ViewChild(MdCard, { read: ViewContainerRef }) card;    
    showSnackbar() {
        this.snackbar.open('UPDATED', '', { viewContainerRef: this.card });
    }
```

I confirmed in the debugger that this.card is defined and assigned a correct ViewContainerRef object.

Once again, I am trying to open the snack bar at the bottom of each card. However, the snack bar opens at the bottom of the browser view port. Am I doing something wrong, or does there seem a to be a bug?

**Comment by jelbourn** [comment]
@cheresier That's not what `viewContainer` is for. Setting the `viewContainer` tells Angular where the loaded component lives in terms of dependency injection and change detection, but does not affect where it is rendered. Per material design spec, the snack bar always renders at the edge of the screen.

**Comment by cheresier** [comment]
@jelbourn, a-ha! Got it. Thank you! Can you point to any other material component I should be using for this use case?

**Comment by jelbourn** [comment]
You may just want make your own component and conditionally show/hide it at the bottom of the card.

**Comment by kuncevic** [comment]
How would you control the position of the snackbar? I can see in the sample it is appearing just in the bottom. How would you display it in the right top corner ?

**Comment by jammerware** [comment]
@kuncevic See [this comment](https://github.com/angular/material2/issues/115#issuecomment-267480428) above. I don't think the position of the snackbar is exposed to the end developer because the material spec doesn't allow it.

**Comment by kuncevic** [comment]
Just went thorugh this https://material.io/guidelines/components/snackbars-toasts.html

> Only one snackbar may be displayed at a time.

> Snackbars animate upwards from the bottom edge of the screen.

**Comment by MikeMatusz** [comment]
On tablet/desktop the material spec does allow for some variation in position:

>Alignment: Centered or left-aligned 24dp from the left and bottom edges of the screen

Are there any plans to support the bottom/left alignment?

**Comment by ethanmick** [comment]
@MikeMatusz brings up a good point, the snackbar can be positioned in the bottom left on Desktop/Tablet. Should there be a new issue for that functionality?

**Comment by mtpultz** [comment]
@MikeMatusz and @ethanmick do you think your last two comments should be moved to a new issue since this issue doesn't actually refer to the position and has been closed.  Wondering if it won't get noticed, but is a case that needs to be addressed.

**Comment by MikeMatusz** [comment]
@mtpultz Good call, I'll create one right now.
